pg_search/src/postgres/customscan/pdbscan/mod.rs

```diff
@@ -995,14 +995,78 @@ impl CustomScan for PdbScan {
         }
 
         // Add a flag to indicate if the query is a full index scan
-        if state
-            .custom_state()
-            .base_search_query_input()
-            .is_full_scan_query()
-        {
-            explainer.add_bool("Full Index Scan", true);
+        let base_query = state.custom_state().base_search_query_input();
+
+        // Only process the query if it's initialized
+        // For EXPLAIN without ANALYZE, the query might not be initialized yet
+        if !matches!(base_query, SearchQueryInput::Uninitialized) {
+            if base_query.is_full_scan_query() {
+                explainer.add_bool("Full Index Scan", true);
+            }
+
+            // Show query with integrated estimates if GUC is enabled and verbose
+            if gucs::explain_recursive_estimates() && explainer.is_verbose() {
+                // Try to get the search reader if available (for actual execution)
+                // For EXPLAIN-only queries, we'll create a temporary reader just for estimates
+                if let Some(search_reader) = state.custom_state().search_reader.as_ref() {
+                    // We have a search reader (this is EXPLAIN ANALYZE)
+                    match search_reader.build_query_tree_with_estimates(base_query.clone()) {
+                        Ok(query_tree) => {
+                            explainer.add_query_with_estimates(&query_tree);
+                        }
+                        Err(e) => {
+                            // Fallback to query without estimates
+                            explainer.add_query(base_query);
+                            explainer.add_text("Estimate Error", format!("{}", e));
+                        }
```

Comment:
IMO, should remove this fallback (and others in this method) and let it panic.

Unless you've seen otherwise for some reason specific to this PR, failures to handle successfully parsed `SearchQueryInput`s are very rare.